superset/views/database/views.py

```diff
@@ -91,11 +91,14 @@ def form_post(self, form):
             return redirect("/csvtodatabaseview/form")
 
         csv_file = form.csv_file.data
-        form.csv_file.data.filename = secure_filename(form.csv_file.data.filename)
+
         csv_filename = form.csv_file.data.filename
-        path = os.path.join(config["UPLOAD_FOLDER"], csv_filename)
+        extension = os.path.splitext(csv_filename)[1].lower()
+        path = tempfile.NamedTemporaryFile(
+            dir=app.config["UPLOAD_FOLDER"], suffix=extension
+        ).name
+        form.csv_file.data.filename = path
         try:
-            utils.ensure_path_exists(config["UPLOAD_FOLDER"])
             csv_file.save(path)
```

Comment:
Where does the file live before this call to save()? In mem?

Reply:
I've seen an OOM issue with uploading "large" files - would be great to buffer these uploaded files to disk as opposed to storing them in mem

Reply:
Did a stream upload with configurable chunks. Yet this may not address the OOM issue if it's related with the pandas dataframe csv loading